### PR TITLE
overload set to accept cache op

### PR DESCRIPTION
**Pull request by liqiangxl**
add another overload of `set` function to accept cache op.
separated from #1173.

### CODE REVIEW

File: csrc/ops/alias.cpp [hunk 1]
```diff
@@ -22,10 +22,20 @@ Val* set(Val* v) {
   return out;
 }
 
+Val* set(Val* v, CacheOp cache_op) {
```
Comment (wujingyue) [inline csrc/ops/alias.cpp:25]: How about giving `cache_op` here the default value `CacheOp::Unspecified` and remove the other flavor of `set` that doesn't take `cache_op`?

**Review by wujingyue — CHANGES_REQUESTED** [review]

**Comment by naoyam** [comment]
Why do we need to have this? I'm a bit nervous about exposing scheduling options through the user-facing interface.

**Comment by liqiangxl** [comment]
> Why do we need to have this? I'm a bit nervous about exposing scheduling options through the user-facing interface.

Just for convenience, allows us to do `auto x1 = set(x, CacheOp::AllLevels);` instead of doing extra code to change the cache op.
```
  TensorView* tv1 =
      ops::newValLike(tv0, tv0->getDataType().value())->as<TensorView>();
  IrBuilder::create<LoadStoreOp>(LoadStoreOpType::Set, tv1, tv0, cache_op);
```
If this is a concern, I can close this PR.

**Comment by naoyam** [comment]
The ops interface is the one we provide to the user and should not have options like this. Otherwise, we would need to deal with user-specified caching options.

**Pull request closed without merging**